lib/cerner/oauth1a/keys.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'openssl'
+require 'uri'
+
+module Cerner
+  module OAuth1a
+    # Public: Keys for authenticating Access Tokens by service providers. Keys can be retrieved
+    # via AccessTokenAgent#retrieve_keys.
+    class Keys
+      # Returns the String version identifier of the keys.
+      attr_reader :version
+      # Returns the String AES secret key.
+      attr_reader :aes_secret_key
+      # Returns the String RSA public key.
+      attr_reader :rsa_public_key
+
+      # Public: Constructs an instance.
+      #
+      # arguments - The keyword arguments of the method:
+      #             :version        - The version identifier of the keys.
+      #             :aes_secret_key - The AES secret key.
+      #             :rsa_public_key - The RSA public key.
+      #
+      # Raises ArgumentError if version, aes_secret_key or rsa_public_key is nil.
+      def initialize(version:, aes_secret_key:, rsa_public_key:)
+        raise ArgumentError, 'version is nil' unless version
+        raise ArgumentError, 'aes_secret_key is nil' unless aes_secret_key
+        raise ArgumentError, 'rsa_public_key is nil' unless rsa_public_key
```

Comment:
Is it possible for these `ArgumentError`s to ever execute? From what I understand about keyword arguments ruby should take care of this for you.

```ruby
>> class Things
     def initialize(version:, aes_secret_key:, rsa_public_key:)
     end
   end
    ==>:initialize

>> Things.new
Traceback (most recent call last):
        4: from /Users/bv031773/.rbenv/versions/2.5.0/bin/irb:11:in `<main>'
        3: from (irb):6
        2: from (irb):6:in `new'
        1: from (irb):2:in `initialize'
ArgumentError (missing keywords: version, aes_secret_key, rsa_public_key)

>> Things.new(version: 'hi')
Traceback (most recent call last):
        4: from /Users/bv031773/.rbenv/versions/2.5.0/bin/irb:11:in `<main>'
        3: from (irb):7
        2: from (irb):7:in `new'
        1: from (irb):2:in `initialize'
ArgumentError (missing keywords: aes_secret_key, rsa_public_key)

>> Things.new(aes_secret_key: 'hi')
Traceback (most recent call last):
        4: from /Users/bv031773/.rbenv/versions/2.5.0/bin/irb:11:in `<main>'
        3: from (irb):8
        2: from (irb):8:in `new'
        1: from (irb):2:in `initialize'
ArgumentError (missing keywords: version, rsa_public_key)

>> Things.new(rsa_public_key: 'hi')
Traceback (most recent call last):
        4: from /Users/bv031773/.rbenv/versions/2.5.0/bin/irb:11:in `<main>'
        3: from (irb):9
        2: from (irb):9:in `new'
        1: from (irb):2:in `initialize'
ArgumentError (missing keywords: version, aes_secret_key)
```

Reply:
Aha, if you pass `nil` explicitly the system falls down.